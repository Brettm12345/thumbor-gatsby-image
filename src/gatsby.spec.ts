import thumbor from "thumbor-ts";

import { fluid, fixed } from "./gatsby";

jest.mock("./remote");

describe("thumbor gatsby image", () => {
  const image = thumbor({
    serverUrl: "http://thumbor",
    imagePath: "example.png",
  });
  test("Fluid", async () => {
    expect(await fluid(image, { maxWidth: 300 })).toMatchSnapshot();
  });
  test("Fixed", async () => {
    expect(await fixed(image, { width: 400, height: 400 })).toMatchSnapshot();
  });
});
