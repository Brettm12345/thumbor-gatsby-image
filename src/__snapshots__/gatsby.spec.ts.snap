// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thumbor gatsby image Fixed 1`] = `
Object {
  "aspectRatio": 1,
  "base64": "",
  "height": 400,
  "src": "http://thumbor/unsafe/400x400/example.png",
  "srcSet": "http://thumbor/unsafe/400x400/example.png 1x, http://thumbor/unsafe/600x600/example.png 1.5x, http://thumbor/unsafe/800x800/example.png 2x, http://thumbor/unsafe/1200x1200/example.png 3x",
  "srcSetWebp": "http://thumbor/unsafe/400x400/filters:format(webp)/example.png 1x, http://thumbor/unsafe/600x600/filters:format(webp)/example.png 1.5x, http://thumbor/unsafe/800x800/filters:format(webp)/example.png 2x, http://thumbor/unsafe/1200x1200/filters:format(webp)/example.png 3x",
  "srcWebp": "http://thumbor/unsafe/400x400/filters:format(webp)/example.png",
  "width": 400,
}
`;

exports[`thumbor gatsby image Fluid 1`] = `
Object {
  "aspectRatio": 0.6666666666666666,
  "base64": "",
  "sizes": "(max-width: 300px) 100vw, 300px",
  "src": "http://thumbor/unsafe/300x0/example.png",
  "srcSet": "http://thumbor/unsafe/75x113/example.png 75w, http://thumbor/unsafe/150x225/example.png 150w, http://thumbor/unsafe/300x450/example.png 300w, http://thumbor/unsafe/450x675/example.png 450w, http://thumbor/unsafe/600x900/example.png 600w, http://thumbor/unsafe/900x1350/example.png 900w",
  "srcSetWebp": "http://thumbor/unsafe/75x113/filters:format(webp)/example.png 75w, http://thumbor/unsafe/150x225/filters:format(webp)/example.png 150w, http://thumbor/unsafe/300x450/filters:format(webp)/example.png 300w, http://thumbor/unsafe/450x675/filters:format(webp)/example.png 450w, http://thumbor/unsafe/600x900/filters:format(webp)/example.png 600w, http://thumbor/unsafe/900x1350/filters:format(webp)/example.png 900w",
  "srcWebp": "http://thumbor/unsafe/300x0/filters:format(webp)/example.png",
}
`;
